tests/integration/cloud/clouds/test_digitalocean.py

```diff
@@ -122,53 +77,33 @@ def test_key_management(self):
             )
 
             # List all keys
-            list_keypairs = self.run_cloud('-f list_keypairs {0}'.format(PROVIDER_NAME))
+            list_keypairs = self.run_cloud('-f list_keypairs {0}'.format(self.PROVIDER))
 
             self.assertIn(
                 finger_print,
                 [i.strip() for i in list_keypairs]
             )
 
             # List key
-            show_keypair = self.run_cloud('-f show_keypair {0} keyname={1}'.format(PROVIDER_NAME,
-                                                                                   do_key_name))
+            show_keypair = self.run_cloud('-f show_keypair {0} keyname={1}'.format(self.PROVIDER, do_key_name))
             self.assertIn(
                 finger_print,
                 [i.strip() for i in show_keypair]
             )
         except AssertionError:
             # Delete the public key if the above assertions fail
-            self.run_cloud('-f remove_key {0} id={1}'.format(PROVIDER_NAME, finger_print))
+            self.run_cloud('-f remove_key {0} id={1}'.format(self.PROVIDER, finger_print))
             raise
         finally:
             # Delete public key
-            self.assertTrue(self.run_cloud('-f remove_key {0} id={1}'.format(PROVIDER_NAME, finger_print)))
+            self.assertTrue(self.run_cloud('-f remove_key {0} id={1}'.format(self.PROVIDER, finger_print)))
 
     def test_instance(self):
         '''
         Test creating an instance on DigitalOcean
         '''
         # check if instance with salt installed returned
-        try:
-            self.assertIn(
-                INSTANCE_NAME,
-                [i.strip() for i in self.run_cloud('-p digitalocean-test {0}'.format(INSTANCE_NAME), timeout=500)]
-            )
-        except AssertionError:
-            self.run_cloud('-d {0} --assume-yes'.format(INSTANCE_NAME), timeout=500)
-            raise
-
-        # delete the instance
-        try:
-            self.assertIn(
-                'True',
-                [i.strip() for i in self.run_cloud('-d {0} --assume-yes'.format(INSTANCE_NAME), timeout=500)]
-            )
-        except AssertionError:
-            raise
+        ret_str = self.run_cloud('-p digitalocean-test {0}'.format(self.instance_name), timeout=TIMEOUT)
+        self.assertInstanceExists(ret_str)
 
-        # Final clean-up of created instance, in case something went wrong.
-        # This was originally in a tearDown function, but that didn't make sense
-        # To run this for each test when not all tests create instances.
-        if INSTANCE_NAME in [i.strip() for i in self.run_cloud('--query')]:
-            self.run_cloud('-d {0} --assume-yes'.format(INSTANCE_NAME), timeout=500)
+        self._destroy_instance()
```

Comment:
Probably, `self.addCleanup(self._destroy_instance)` before calling the assert method. This way, even if the assertion method fails, the cleanup code will run.

Reply:
the tearDown inherited from the CloudTest metaclass will make sure that `self._destroy_instance()` is run.  It will also fail the test if the instance is destroyed from the tearDown because it should be successful the first time `self._destroy_instance()` is called